File: Frontend/construction-project-tracker/src/app/core/models/project.ts
import { ProjectStatus } from '../enums/project-status';

export interface Project {
  id: number;
  name: string;
  description: string;
  budget: number;
  startDate: string;
  endDate: string;
  status: ProjectStatus;
  progressPercentage: number;
}

export interface ProjectDetails extends Project {
  assignedEngineersCount: number;
  tasksCount: number;
  documentsCount: number;
}

export interface CreateProject {
  name: string;
  description: string;
  budget: number;
  startDate: string;
  endDate: string;
}

export interface UpdateProject extends CreateProject {}

File: Backend/ConstructionProjectTracker.API/Entities/Project.cs
using ConstructionProjectTracker.API.Enums;

namespace ConstructionProjectTracker.API.Entities;

public class Project
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Budget { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public ProjectStatus Status { get; set; }
    public int ProgressPercentage { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<ProjectAssignment> ProjectAssignments { get; set; } = new List<ProjectAssignment>();
    public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
    public ICollection<Document> Documents { get; set; } = new List<Document>();
}
